### ISSUE TITLE
Adding PhysFormer pipeline

### ISSUE THREAD

**Issue by Anshul22Verma** [Anshul22Verma/rPPG#1]
Fredie to work on adding PhysFormer pipeline to replicate all the experiments.

**Comment by RuileLu** [comment]
Working on (and is stuck on) figuring out how the ECG labels are produced during preprocessing for each sample in the VIPL_HR dataset.

**Comment by RuileLu** [comment]
pushed(last week) the code for other preprocessing steps (need to add down sampling as well).

**Comment by RuileLu** [comment]
Sent an email to the author(physformer paper) to ask about how ECG labels are produced from raw VIPL_HR data(yesterday) but did not hear back. Hey Anshul, if you have time could you take a look? Thanks!

**Comment by Anshul22Verma** [comment]
Hey Fredie, I will try to look at tomorrow evening and let you know if I find anything.

**Comment by RuileLu** [comment]
Hey Ashul, in the loadtemporal_data file VIPL_train class, ecg label for each sample is extracted from the preprocessed landmarks_frame file(line109) as instances of length 160. They are then after modification(get_single_video_x_aug) or not (get_single_video_x) used in train_physformer_160_VIPL as labels(line 191). However with each sample folder of raw data in VIPL_HR, I could only find a gt_HR, a gt_SpO2, a time file, a bvp wave file, and the video. I am not sure how the ecg labels are derived. OR if it' actually bvp label, how is the preprocessing is done from bvp wave given?  Thanks!

**Comment by RuileLu** [comment]
For the exact format/sample of ecg label, can refer to VIPL_folder1_train(many) and VIPL_folder1_test1 file(1). I checked the preprocessing part of the paper there is no mention of this. I'll look into the model part more too, to see if it's mentioned in the model section

**Comment by RuileLu** [comment]
Hey Anshul, I was working on a school project these two days. I will try to finish running the script with sample data tomorrow so that we can start training the physformer the day after : ) , sorry about the delay! lemme know if you figured out how the ecg to rppg conversion works, I will use simulated rppg for now.

**Comment by RuileLu** [comment]
Done training a sample(p0) and testing it on the provided one testing sample:) Will upload the code once finish cleaning up the code. Will start writing the training script for cluster.

**Comment by Anshul22Verma** [comment]
That is just amazing

**Comment by RuileLu** [comment]
Pushed the physformer code
